pkg/scheduler/objects/node.go

```diff
@@ -316,15 +316,17 @@ func (sn *Node) AddAllocation(alloc *Allocation) bool {
 	return false
 }
 
-// Replace the paceholder allocation on the node. No usage changes as the placeholder must
-// be the same size as the real allocation.
-func (sn *Node) ReplaceAllocation(uuid string, replace *Allocation) {
+// ReplaceAllocation replaces the placeholder with the real allocation on the node.
+// The delta passed in is the difference in resource usage between placeholder and real allocation.
+// It should always be a negative value (really a decreased usage).
```

Comment:
If this is a strong requirement, then it might make sense to at least log a warning if this expectation (`delta.HasNegativeValue() == true`) is not met. This can lead to bugs that are hard to figure out.

Reply:
added log entry